### ISSUE TITLE
MNT Deal with pandas 2.0 vs 2.1

### ISSUE THREAD

**Issue by riedgar-ms** [fairlearn/fairlearn#1299]
In `pandas` v2.1, a new `DataFrame.map()` API was added, while the previous `DataFrame.applymap()` was simultaneously deprecated. Since our builds fail on warnings, this causes issues if the latest `pandas` cannot be install (notably for Python 3.8, but also some of our 'otherml_packages' tests).

PR #1302  squares this particular circle by rewriting the handful of `DataFrame.applymap()` calls to being nested `DataFrame.apply()` ones. This is not going to be particularly performant, but the DataFrames in question should be relatively small.

This issue is to track the nested 'apply' calls, so they can be improved once we drop `pandas` v2.0.

*Note* This has been edited, so some of the discussion below may be obsolete.

**Comment by adrinjalali** [comment]
The issue with not handling this now, is that downstream libraries are getting the `FutureWarning` and their CI fails due to that (I work on one of those libraries)

**Comment by riedgar-ms** [comment]
Sorry.... `FutureWarning` from what? pandas?

**Comment by adrinjalali** [comment]
Yes, the warning is propagated and it's making my CI fail in `skops`.

**Comment by adrinjalali** [comment]
So when would people stop getting the future warnings coming from pandas? Are you saying we'd fix those warnings somehow in the next release, or the one after? If the one after, it'd mean people would get those warnings for quite a long time from our side.

**Comment by riedgar-ms** [comment]
And for extra fun, pandas 2.2 is now out and wreakings its havoc. Some fixes in #1351

**Comment by TamaraAtanasoska** [comment]
Closing because of #1522
